packages/flutter/lib/src/material/tabs.dart

```diff
@@ -145,10 +153,14 @@ class Tab extends StatelessWidget implements PreferredSizeWidget{
 
   @override
   Size get preferredSize {
-    if ((text != null || child != null) && icon != null)
-      return const Size.fromHeight(_kTextAndIconTabHeight);
+    final double tabHeight;
+    if (height != null)
+      tabHeight = height!;
+    else if ((text != null || child != null) && icon != null)
+      tabHeight = _kTextAndIconTabHeight;
     else
-      return const Size.fromHeight(_kTabHeight);
+      tabHeight = _kTabHeight;
+    return Size.fromHeight(tabHeight);
```

Comment:
You could just make the height != null an early return for simplicity.

Reply:
Done